src/clr-angular/utils/popover/adapter/operators.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { ClrPopoverPosition } from '../interfaces/popover-position.interface';
+import { ClrAxis } from '../enums/axis.enum';
+import { ClrSide } from '../enums/side.enum';
+import { ClrAlignment } from '../enums/alignment.enum';
+
+export const SIMPLE_TO_SMART_POSITIONS: { [input: string]: ClrPopoverPosition } = {
+  'top-left': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.END,
+  },
+  'top-middle': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.CENTER,
+  },
+  'top-right': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.START,
+  },
+  'right-top': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.END,
+  },
+  'right-middle': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.CENTER,
+  },
+  'right-bottom': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.START,
+  },
+  'bottom-right': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.START,
+  },
+  'bottom-middle': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.CENTER,
+  },
+  'bottom-left': {
+    axis: ClrAxis.VERTICAL,
+    side: ClrSide.AFTER,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.END,
+  },
+  'left-bottom': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.START,
+  },
+  'left-middle': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.CENTER,
+  },
+  'left-top': {
+    axis: ClrAxis.HORIZONTAL,
+    side: ClrSide.BEFORE,
+    anchor: ClrAlignment.CENTER,
+    content: ClrAlignment.END,
+  },
+};
+
+export function posiblePositionsAsStrings() {
+  return Object.keys(SIMPLE_TO_SMART_POSITIONS);
+}
+
+export function defaultPosition() {
```

Comment:
I don't think this name will work. Default position for dropdowns is `bottom-left`. 
It seems like the component should be responsible for setting the default string. Not here.

Reply:
Ok but let us return `null` when we could not find the position from 
 * `transposeStringPosition`
 * `transposePositionString`

And everyone will have to handle the default state by themself.

Reply:
`requestedPosition` ?

Bozhidar, you have point... now one returns null, the other undefined. Returning null should be the preferred one.

Reply:
Done now both methods are returning `null` as standard response when no position is found